src/datalink/mod.rs

```diff
@@ -229,6 +229,15 @@ impl NetworkInterface {
     }
 }
 
+/// Represents an IP address and subnet mask
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct IpNetmask {
+    /// IP address
+    pub ip: IpAddr,
+    /// Subnet mask
+    pub netmask: Option<IpAddr>,
```

Comment:
Just checked, and it seems that `Option` is appropriate:

> The ifa_netmask field points to a structure containing the netmask associated with ifa_addr, if applicable for the address family.  This field may contain a null pointer.

http://man7.org/linux/man-pages/man3/getifaddrs.3.html